packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -1091,11 +1092,8 @@ Future<DateTimeRange?> showDateRangePicker({
   TextInputType keyboardType = TextInputType.datetime,
   final Icon? switchToInputEntryModeIcon,
   final Icon? switchToCalendarEntryModeIcon,
+  SelectableDayForRangePredicate? selectableDayPredicate,
 }) async {
-  assert(
-    initialDateRange == null || !initialDateRange.start.isAfter(initialDateRange.end),
-    "initialDateRange's start date must not be after it's end date.",
-  );
```

Comment:
Should `selectableDayPredicate` have a default value that would check for this? What happens if `selectableDayPredicate` is null and the start date is after the end date?

Reply:
I've removed this assertion as it is already checked by the `DateTimeRange` itself.